thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyTimeBasedSummarizer.java

```diff
@@ -74,18 +78,21 @@ public int compare(RawAnomalyResultDTO o1, RawAnomalyResultDTO o2) {
         populateMergedResult(mergedAnomaly, currentResult);
       } else {
         // compare current with merged and decide whether to merge the current result or create a new one
-        MergedAnomalyResultDTO currMergedAnomaly = new MergedAnomalyResultDTO();
-        populateMergedResult(currMergedAnomaly, currentResult);
+        MergedAnomalyResultDTO currAnomaly = new MergedAnomalyResultDTO();
+        populateMergedResult(currAnomaly, currentResult);
+        // if the merging is applying sequential gap and current anomaly has gap time larger than sequentialAllowedGap
+        // or current anomaly is not equal on mergeable keys with mergedAnomaly
+        // should not merge the two and split from here
         if ((applySequentialGapBasedSplit
             && (currentResult.getStartTime() - mergedAnomaly.getEndTime()) > sequentialAllowedGap)
-            || (!isMergeable(mergedAnomaly, currMergedAnomaly))) {
+            || (!isEqualOnMergeableKeys(mergedAnomaly, currAnomaly, mergeablePropertyKeys))) {
```

Comment:
you may want to define a new variable name or function to wrapper the if condition:
isNotMergable(*)